sequencer/src/bin/marketplace-solver.rs

```diff
@@ -0,0 +1,90 @@
+use std::{str::FromStr, sync::Arc};
+
+use anyhow::Context;
+use async_compatibility_layer::logging::{setup_backtrace, setup_logging};
+use async_std::{sync::RwLock, task::spawn};
+use clap::Parser;
+use espresso_types::SeqTypes;
+use hotshot_types::traits::node_implementation::NodeType;
+use marketplace_solver::{
+    define_api, handle_events,
+    state::{GlobalState, SolverState, StakeTable},
+    DatabaseOptions, EventsServiceClient, SolverError,
+};
+use tide_disco::App;
+use url::Url;
+use vbs::version::StaticVersionType;
+
+type Version = <SeqTypes as NodeType>::Base;
+
+#[derive(Parser)]
+struct Args {
+    /// Port to run the server on.
+    #[clap(short, long, env = "ESPRESSO_MARKETPLACE_SOLVER_API_PORT")]
+    solver_api_port: u16,
+
+    /// Hotshot events service api URL
+    #[clap(short, long, env = "ESPRESSO_SEQUENCER_HOTSHOT_EVENT_API_URL")]
+    events_api_url: String,
+
+    #[clap(flatten)]
+    database_options: DatabaseOptions,
```

Comment:
I might add the database options here in the future, instead of importing them from the marketplace-solver crate, to keep the env names consistent.